WalletWasabi/BitcoinCore/Monitoring/P2pReconnector.cs

```diff
@@ -23,12 +23,12 @@ protected override async Task ActionAsync(CancellationToken cancel)
 			try
 			{
 				Logger.LogInfo("Trying to reconnect to P2P...");
-				P2pNode.Disconnect();
+				await P2pNode.DisconnectAsync(cancel).ConfigureAwait(false);
```

Comment:
What about making `P2pNode.DisconnectAsync` return `bool` whether it succeeded or not? And if it succeeded, then call `P2pNode.ConnectAsync`.

It would be great to add `LogTrace` to make debugable in the future.

Reply:
ACK, @molnard, @kiminuo's concept would also resolve this conversation: https://github.com/zkSNACKs/WalletWasabi/pull/4707/files/9f55fa1fedb9d10e10f4e0ce1412c75452877157#r520763464

Reply:
ACK